newegg/communicator.py

```diff
@@ -79,38 +84,52 @@ def get_transaction_number(self, session_id: str, **kwargs) -> Optional[int]:
+        headers = self.get_default_headers(
+            {'x-sessionid': session_id, 'referer': f'https://secure.newegg.com/shop/checkout?sessionId={session_id}'})
+        data = {'TransNumber': transaction_number, 'AddressLabel': 'Untitled', 'ContactWith': name, 'Phone': phone,
+                'Fax': '', 'Country': country, 'State': state, 'City': city, 'Address1': address, 'Address2': '',
+                'ZipCode': zip_code, 'IsDefault': False,
+                'DisplayLines': [address, f'{city}, {state} {zip_code}', country_long, phone],
+                'AddressVerifyMark': 'Verified', 'Email': None, 'DisableEmail': False, 'CompanyName': '',
+                'LanguageCode': None, 'IsSelected': False, 'SaveAddress': False,
+                'QASDisplayLines': [address, f'{city}, {state} {zip_code}', country_long]}
         try:
             req = requests.post(url, headers=headers, json=data, cookies=self.cookies, timeout=self.timeout)
             DebugLogger.log(f'Validate address text: {req.text}')
             DebugLogger.log(f'Validate address status: {req.status_code}')
-            return req.status_code == 200
+            return req.json()["Result"] == "Success"
         except:
             return False
 
     def submit_order(self, session_id: str, **kwargs) -> bool:
         url = 'https://secure.newegg.com/shop/api/PlaceOrderApi'
-        headers = self.get_default_headers({'x-sessionid': session_id, 'referer': f'https://secure.newegg.com/shop/checkout?sessionId={session_id}'})
-        data = {'SessionID': session_id, 'IsAcceptNSCCAuth': False, 'SubscribeNewsletterFlag': False, 'CreactAccount': False, 'Password': '', 'MobileSubscribe': {}, 'LanguageCode': 'en-us', 'Cvv2': ''}
+        headers = self.get_default_headers(
+            {'x-sessionid': session_id, 'referer': f'https://secure.newegg.com/shop/checkout?sessionId={session_id}'})
+        data = {'SessionID': session_id, 'IsAcceptNSCCAuth': False, 'SubscribeNewsletterFlag': False,
+                'CreactAccount': False, 'Password': '', 'MobileSubscribe': {}, 'LanguageCode': 'en-us', 'Cvv2': ''}
         try:
             req = requests.post(url, headers=headers, json=data, cookies=self.cookies, timeout=self.timeout)
             DebugLogger.log(f'Submit order text: {req.text}')
             DebugLogger.log(f'Submit order status: {req.status_code}')
-            return req.status_code == 200
+            return req.json()["Result"] == "Success"
```

Comment:
I think req.status_code == 200 and req.json()[["result"] check would be more comprehensive, could prevent parsing json() when we dont need to if its a 400 error.